[scorecard] Skip stable-diffusion tests

driazati opened the pull request:
These are both currently failing in CI:
https://gitlab.com/octoml/relax-scorecard-ci2/-/pipelines/835704862

This also bumps the timeout to 2 hours (from 1 hour) to account for
increased runtime due to tuning

YuchenJin commented:
As we discussed on this week's relax open dev meeting, we agreed on having `output_shape` as the 3rd argument of `call_tir`. And for the TVMScript printing sugar, we also converged on the following format, that is we don't print the shape, dtype info on the lhs Var of a binding if the rhs is a `call_tir`, and for the lhs, print the output shape/dtype in the end.
```python
gv = relax.call_tir(batch_norm, (data, bn_data_gamma, bn_data_beta, bn_data_moving_mean, bn_data_moving_var), ((1, 3, 224, 224), (3,), (3,)), dtype=("float32", "float32", "float32"))
```

I made the changes, it would be great to have you guys take a review. cc @yongwww @Hzfengsy @hypercubestart @ZihengJiang. :)

hypercubestart reviewed (commented):
overall looks good! just a few questions

YuchenJin commented on src/relax/ir/expr_functor.cc:
```diff
@@ -221,24 +221,7 @@ TVM_REGISTER_GLOBAL("relax.analysis.post_order_visit").set_body_typed([](Expr ex
 // ==================
 // ExprMutator
 
-Expr ExprMutator::VisitExpr_(const ConstantNode* op) {
```
That's a great question! I should have explained about this change in the pr. I plan to have a formal doc about the shape deduction rule, but currently I have a rough sketch in the discussion thread #88.

The rule is `shape_` and `checked_type_` of Expr can only be deduced, but cannot be modified by the mutator. For this specific case, if the user overrides the `ShapeExprNode` visitor, and here if we don't delete visiting the ConstantNode shape in the visitor, it will visit the constant node's `shape_` field and mutates its `shape_`. While the data in the ConstantNode is unchanged, but we mutates its `shape_`, which is definitely wrong.

YuchenJin commented on python/tvm/relax/op/base.py:
```diff
@@ -14,42 +14,49 @@
     args: Union[Tuple, List[Expr]],
-    tir_vars: ShapeExpr = None,
+    output_shape: Union[Tuple, ShapeExpr, List[int]],
+    output_type: Union[TupleType, DynTensorType],
```
Yeah, I think it's a good suggestion! I will make the change.

YuchenJin replied:
Updated, please take a look. :)

YuchenJin commented on python/tvm/script/relax/parser.py:
```diff
@@ -1020,13 +1021,66 @@ def parse_call(self, expr: ast.Call) -> Union[tir.PrimExpr, relax.Expr]:
                         f"{op.name} expects {op.num_inputs} arguments but got {len(args)}",
                         expr.span,
                     )
+
```
The dtype is a keyword argument (because the out_type in call_tir is a type_arg instead of a argument), so the len(args) should still be either 3 and 4. I do agree that we don't want to use hard-coded valued, so updated it. Thanks for the suggestion!

YuchenJin commented on tests/python/relax/test_vm.py:
```diff
@@ -551,6 +551,36 @@ def te_func(A, B):
 
 
+def test_vm_emit_te_dtype_change():
+    bb = relax.BlockBuilder()
```
Applied a pass and check the dtype in the attribute to make sure the storage allocated is for dtype int16.

YuchenJin commented on python/tvm/relax/block_builder.py:
```diff
@@ -363,7 +364,7 @@ def te_func(var_rxplaceholder: T.handle, var_rxplaceholder_1: T.handle,
                 def rx_func(x: Tensor[(n, m), "float32"], y: Tensor[(n, m), "float32"]) -> Tensor:
                     # block 0
-                    gv = relax.call_tir((128, 128), "te_func", (x, y))
+                    gv = relax.call_tir((128, 128), "te_func", (x, y), dtype="float32")
```
Good catch! Updated, thanks Yong!

hypercubestart reviewed (approved):
LGTM! thanks for all the work!

yongwww commented:
Thanks @YuchenJin for the work and leading the discussion! Thanks folks for reviewing! @ZihengJiang @hypercubestart

driazati committed:
[scorecard] Skip stable-diffusion tests

These are both currently failing in CI:
https://gitlab.com/octoml/relax-scorecard-ci2/-/pipelines/835704862

This also bumps the timeout to 2 hours (from 1 hour) to account for
increased runtime due to tuning
